File: solutions/0026-remove-duplicates-from-sorted-array/solution.java
class Solution {
    public static int[] swap (int[] arr, int index1, int index2){
        int temp = arr[index1];
        arr[index1] = arr[index2];
        arr[index2] = temp;
        return arr;
    }
    public int removeDuplicates(int[] nums) {
       int lastUniqueIndex = 0;
        for(int i = 1;i< nums.length;i++){
            if(nums[i] != nums[lastUniqueIndex]){
//                ++lastUniqueIndex;
                nums = swap(nums, i, lastUniqueIndex+1);
                lastUniqueIndex++;
            }
        }
        // System.out.println(Arrays.toString(nums));
        return lastUniqueIndex+1; 
    }
}

File: solutions/0026-remove-duplicates-from-sorted-array/solution.cpp
class Solution {
public:
    void swap(vector<int>& nums, int index1, int index2){
        int temp = nums[index1];
        nums[index1] = nums[index2];
        nums[index2] = temp;
    }
    int removeDuplicates(vector<int>& nums) {
        int lu = 0;
        for(int i = 1;i<nums.size();i++){
            if(nums[i]>nums[lu]){
                swap(nums, i, lu+1);
                lu = lu+1;
            }
        }
        return lu+1;
    }
};
